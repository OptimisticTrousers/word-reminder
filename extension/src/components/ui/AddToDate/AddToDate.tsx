import CSSModules from "react-css-modules";

import styles from "./AddToDate.module.css";

interface DefaultValues {
  minutes: number;
  hours: number;
  days: number;
  weeks: number;
  months: number;
}

interface Props {
  disabled: boolean;
  legend: string;
  defaultValues: DefaultValues;
}

export const AddToDate = CSSModules(
  function ({ disabled, legend, defaultValues}: Props) {
    const legendLower = legend.toLowerCase();
    return (
      <fieldset styleName="add-to-date" disabled={disabled}>
        <legend styleName="add-to-date__fieldset">{legend}</legend>
        <div styleName="add-to-date__control">
          <label
            styleName="add-to-date__label"
            htmlFor={`${legendLower}-minutes`}
          >
            Minutes
          </label>
          <input
            styleName="add-to-date__input"
            type="number"
            id={`${legendLower}-minutes`}
            name={`${legendLower}-minutes`}
            required
            max={60}
            min={0}
            defaultValue={defaultValues.minutes}
          />
        </div>
        <div styleName="add-to-date__control">
          <label
            styleName="add-to-date__label"
            htmlFor={`${legendLower}-hours`}
          >
            Hours
          </label>
          <input
            styleName="add-to-date__input"
            type="number"
            id={`${legendLower}-hours`}
            name={`${legendLower}-hours`}
            required
            max={24}
            min={0}
            defaultValue={defaultValues.hours}
          />
        </div>
        <div styleName="add-to-date__control">
          <label styleName="add-to-date__label" htmlFor={`${legendLower}-days`}>
            Days
          </label>
          <input
            styleName="add-to-date__input"
            type="number"
            id={`${legendLower}-days`}
            name={`${legendLower}-days`}
            required
            max={31}
            min={0}
            defaultValue={defaultValues.days}
          />
        </div>
        <div styleName="add-to-date__control">
          <label
            styleName="add-to-date__label"
            htmlFor={`${legendLower}-weeks`}
          >
            Weeks
          </label>
          <input
            styleName="add-to-date__input"
            type="number"
            id={`${legendLower}-weeks`}
            name={`${legendLower}-weeks`}
            required
            max={52}
            min={0}
            defaultValue={defaultValues.weeks}
          />
        </div>
        <div styleName="add-to-date__control">
          <label
            styleName="add-to-date__label"
            htmlFor={`${legendLower}-months`}
          >
            Months
          </label>
          <input
            styleName="add-to-date__input"
            type="number"
            id={`${legendLower}-months`}
            name={`${legendLower}-months`}
            required
            max={12}
            min={0}
            defaultValue={defaultValues.months}
          />
        </div>
      </fieldset>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "log" }
);
