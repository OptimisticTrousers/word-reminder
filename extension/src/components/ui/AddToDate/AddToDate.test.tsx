import { render, screen } from "@testing-library/react";
import { AddToDate } from "./AddToDate";

describe("AddToDate", () => {
  it("renders five inputs for minutes, hours, days, weeks, and months", async () => {
    const { asFragment } = render(
      <AddToDate
        disabled={false}
        legend="Reminder"
        defaultValues={{
          minutes: 5,
          hours: 2,
          days: 3,
          weeks: 4,
          months: 1,
        }}
      />
    );

    const legend = screen.getByText("Reminder");
    const minutesInput = screen.getByLabelText("Minutes");
    const hoursInput = screen.getByLabelText("Hours");
    const daysInput = screen.getByLabelText("Days");
    const weeksInput = screen.getByLabelText("Weeks");
    const monthsInput = screen.getByLabelText("Months");

    expect(legend).toBeInTheDocument();
    expect(minutesInput).toBeInTheDocument();
    expect(minutesInput).toHaveAttribute("name", "reminder-minutes");
    expect(minutesInput).toHaveValue(5);
    expect(minutesInput).not.toBeDisabled();
    expect(hoursInput).toBeInTheDocument();
    expect(hoursInput).toHaveAttribute("name", "reminder-hours");
    expect(hoursInput).toHaveValue(2);
    expect(hoursInput).not.toBeDisabled();
    expect(daysInput).toBeInTheDocument();
    expect(daysInput).toHaveAttribute("name", "reminder-days");
    expect(daysInput).toHaveValue(3);
    expect(daysInput).not.toBeDisabled();
    expect(weeksInput).toBeInTheDocument();
    expect(weeksInput).toHaveAttribute("name", "reminder-weeks");
    expect(weeksInput).toHaveValue(4);
    expect(weeksInput).not.toBeDisabled();
    expect(monthsInput).toBeInTheDocument();
    expect(monthsInput).toHaveAttribute("name", "reminder-months");
    expect(monthsInput).toHaveValue(1);
    expect(monthsInput).not.toBeDisabled();
    expect(asFragment()).toMatchSnapshot();
  });

  it("disables all five inputs", async () => {
    render(
      <AddToDate
        disabled={true}
        legend="Duration"
        defaultValues={{
          minutes: 0,
          hours: 0,
          days: 0,
          weeks: 0,
          months: 0,
        }}
      />
    );

    const legend = screen.getByText("Duration");
    const minutesInput = screen.getByLabelText("Minutes");
    const hoursInput = screen.getByLabelText("Hours");
    const daysInput = screen.getByLabelText("Days");
    const weeksInput = screen.getByLabelText("Weeks");
    const monthsInput = screen.getByLabelText("Months");

    expect(legend).toBeInTheDocument();
    expect(minutesInput).toBeInTheDocument();
    expect(minutesInput).toBeDisabled();
    expect(hoursInput).toBeInTheDocument();
    expect(hoursInput).toBeDisabled();
    expect(daysInput).toBeInTheDocument();
    expect(daysInput).toBeDisabled();
    expect(weeksInput).toBeInTheDocument();
    expect(weeksInput).toBeDisabled();
    expect(monthsInput).toBeInTheDocument();
    expect(monthsInput).toBeDisabled();
  });
});
