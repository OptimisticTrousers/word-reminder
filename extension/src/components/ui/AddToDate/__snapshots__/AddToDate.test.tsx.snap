// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddToDate > renders five inputs for minutes, hours, days, weeks, and months 1`] = `
<DocumentFragment>
  <fieldset
    class="_add-to-date_52612a"
  >
    <legend
      class="_add-to-date__fieldset_52612a"
    >
      Reminder
    </legend>
    <div
      class="_add-to-date__control_52612a"
    >
      <label
        class="_add-to-date__label_52612a"
        for="reminder-minutes"
      >
        Minutes
      </label>
      <input
        class="_add-to-date__input_52612a"
        id="reminder-minutes"
        max="60"
        min="0"
        name="reminder-minutes"
        required=""
        type="number"
        value="5"
      />
    </div>
    <div
      class="_add-to-date__control_52612a"
    >
      <label
        class="_add-to-date__label_52612a"
        for="reminder-hours"
      >
        Hours
      </label>
      <input
        class="_add-to-date__input_52612a"
        id="reminder-hours"
        max="24"
        min="0"
        name="reminder-hours"
        required=""
        type="number"
        value="2"
      />
    </div>
    <div
      class="_add-to-date__control_52612a"
    >
      <label
        class="_add-to-date__label_52612a"
        for="reminder-days"
      >
        Days
      </label>
      <input
        class="_add-to-date__input_52612a"
        id="reminder-days"
        max="31"
        min="0"
        name="reminder-days"
        required=""
        type="number"
        value="3"
      />
    </div>
    <div
      class="_add-to-date__control_52612a"
    >
      <label
        class="_add-to-date__label_52612a"
        for="reminder-weeks"
      >
        Weeks
      </label>
      <input
        class="_add-to-date__input_52612a"
        id="reminder-weeks"
        max="52"
        min="0"
        name="reminder-weeks"
        required=""
        type="number"
        value="4"
      />
    </div>
    <div
      class="_add-to-date__control_52612a"
    >
      <label
        class="_add-to-date__label_52612a"
        for="reminder-months"
      >
        Months
      </label>
      <input
        class="_add-to-date__input_52612a"
        id="reminder-months"
        max="12"
        min="0"
        name="reminder-months"
        required=""
        type="number"
        value="1"
      />
    </div>
  </fieldset>
</DocumentFragment>
`;
